libs/service-portal/access-control/delegations/src/screens/AccessControl/AccessControl.tsx

```diff
@@ -0,0 +1,152 @@
+import { Box, Text, Tabs } from '@island.is/island-ui/core'
+import {
+  AccessDenied,
+  IntroHeader,
+  m,
+  NoDataScreen,
+  ServicePortalModuleComponent,
+  ServicePortalPath,
+} from '@island.is/service-portal/core'
+
+import { Accesses } from '../../components'
+import { useLocale, useNamespaces } from '@island.is/localization'
+import { gql, useQuery } from '@apollo/client'
+import { Query } from '@island.is/api/schema'
+import { useAuth } from '@island.is/auth/react'
+import { MessageDescriptor } from 'react-intl'
+import { useHistory, useLocation } from 'react-router-dom'
+
+export const AuthDelegationsQuery = gql`
+  query AuthDelegationsListQuery {
+    authDelegations {
+      ... on AuthCustomDelegation {
+        validTo
+      }
+    }
+  }
+`
+
+const AccessControl: ServicePortalModuleComponent = ({ userInfo, client }) => {
+  useNamespaces('sp.settings-access-control')
+  const { data, loading } = useQuery<Query>(AuthDelegationsQuery)
+  const { switchUser } = useAuth()
+  const { formatMessage } = useLocale()
+  const history = useHistory()
+  const location = useLocation()
+  const isDelegationToMe =
+    location.pathname === ServicePortalPath.AccessControlDelegationsToMe
+
+  if (!loading && data?.authDelegations.length === 0) {
+    return (
+      <NoDataScreen
+        title={formatMessage({
+          id: 'sp.settings-access-control:empty-title',
+          defaultMessage: 'Umboð',
+        })}
+        button={{
+          type: 'internal',
+          link: ServicePortalPath.AccessControlDelegationsGrant,
+          text: formatMessage({
+            id: 'sp.settings-access-control:empty-new-access',
+            defaultMessage: 'Veita aðgang',
+          }),
+          variant: 'primary',
+        }}
+        secondaryButton={{
+          type: 'click',
+          onClick: () => switchUser(),
+          text: formatMessage({
+            id: 'sp.settings-access-control:empty-switch-access',
+            defaultMessage: 'Skipta um notanda',
+          }),
+          variant: 'ghost',
+        }}
+      >
+        <Text>
+          {formatMessage({
+            id: 'sp.settings-access-control:empty-intro',
+            defaultMessage:
+              'Hérna kemur listi yfir þau umboð sem þú hefur gefið öðrum. Þú getur eytt umboðum eða bætt við nýjum.',
+          })}
+        </Text>
+      </NoDataScreen>
+    )
+  }
+  const actor = userInfo.profile.actor
+  const isDelegation = Boolean(actor)
+
+  const isCompany = userInfo.profile.subjectType === 'legalEntity'
+  const personDelegation = isDelegation && !isCompany
+
+  if (personDelegation) {
+    return <AccessDenied userInfo={userInfo} client={client} />
+  }
```

Comment:
Both of these early returns should probably be inside the tab screen component? Also the data fetching logic? And we should return access denied before no data screen (probably not your code i know).

Reply:
fixed